include/gzip/compress.hpp

```diff
@@ -14,13 +14,27 @@ class Compressor
 {
     std::size_t max_;
     int level_;
+    struct libdeflate_compressor* compressor_ = nullptr;
 
   public:
-    Compressor(int level = Z_DEFAULT_COMPRESSION,
+    Compressor(int level = 6,
                std::size_t max_bytes = 2000000000) // by default refuse operation if uncompressed data is > 2GB
         : max_(max_bytes),
           level_(level)
     {
+        compressor_ = libdeflate_alloc_compressor(level_);
+        if (!compressor_)
+        {
+            throw std::runtime_error("libdeflate_alloc_compressor failed");
+        }
+    }
+
+    ~Compressor()
+    {
+        if (compressor_)
+        {
+            libdeflate_free_compressor(compressor_);
+        }
```

Comment:
Noting that this approach (initialize C struct pointer in constructor + free the memory in the deconstructor) is applying RAII (https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) to avoid a memory leak.